messaging/src/main/java/org/axonframework/common/property/AbstractMethodPropertyAccessStrategy.java

```diff
@@ -37,20 +39,33 @@ public abstract class AbstractMethodPropertyAccessStrategy extends PropertyAcces
     @Override
     public <T> Property<T> propertyFor(Class<? extends T> targetClass, String property) {
         String methodName = getterName(property);
-        try {
-            final Method method = targetClass.getMethod(methodName);
-            if (!Void.TYPE.equals(method.getReturnType())) {
-                return new MethodAccessedProperty<>(method, property);
-            }
-            logger.debug(
-                    "Method with name '{}' in '{}' cannot be accepted as a property accessor, as it returns void",
-                    methodName, targetClass.getName());
-        } catch (NoSuchMethodException e) {
+        Optional<Method> method = getMethod(targetClass, methodName);
+        if (!method.isPresent()) {
             logger.debug("No method with name '{}' found in {} to use as property accessor. " +
                                  "Attempting to fall back to other strategies.",
                          methodName, targetClass.getName());
+            return null;
+        } else {
+            return new MethodAccessedProperty<>(method.get(), property);
+        }
+    }
+
+    private <T> Optional<Method> getMethod(Class<T> targetClass, String methodName) {
+        return Arrays.stream(targetClass.getMethods())
+                     .filter(method -> method.getName().equals(methodName))
+                     .filter(method -> method.getParameterCount() == 0)
```

Comment:
Although the parameter count check is new, after giving it some thought, it makes sense in the scope of a `PropertyAccessStrategy`. 

Added, since the API doesn't provide the option to give the parameters to access a property *if* that method requires them signals it is not intended to work with a parameter count higher than zero.

Reply:
nothing todo for me, right?

Reply:
No, it looks fine. The parameter check was implicit, as `class.getMethod` has a vararg of parameter types. The list was empty, indicating a no-arg method.